pulsar-broker/src/test/java/org/apache/pulsar/client/impl/RawReaderTest.java

```diff
@@ -226,4 +232,44 @@ public void testFlowControl() throws Exception {
         Assert.assertEquals(timeouts, 1);
         Assert.assertEquals(keys.size(), numMessages);
     }
+
+    @Test
+    public void testAcknowledgeWithProperties() throws Exception {
+        int numKeys = 10;
+
+        String topic = "persistent://my-property/use/my-ns/my-raw-topic";
+
+        Set<String> keys = publishMessages(topic, numKeys);
+
+        MessageId lastMessageId = null;
+        RawReader reader = RawReader.create(pulsarClient, subscription, topic).get();
+        try {
+            while (true) { // should break out with TimeoutException
+                try (RawMessage m = reader.readNextAsync().get(1, TimeUnit.SECONDS)) {
```

Comment:
is it better for use for-loop `for (int i = 0; i < numKeys; i++)` rather than using a `while (true)` loop and waiting for timeout? It is adding 1 second here.

Reply:
I also want to verify that we read only what was written, and nothing more. I guess I could reduce the timeout to 100ms or something.